flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/SupportsShowLikeOperation.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.operations;
+
+import org.apache.flink.table.operations.utils.OperationLikeType;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/** Abstract show operation supports filter with like. */
+public abstract class SupportsShowLikeOperation implements ShowOperation {
+
+    // different like type such as like, ilike
+    private final OperationLikeType likeType;
+    private final boolean notLike;
+    private final String likePattern;
+
+    /** Use when there is no sub-clause. */
+    protected SupportsShowLikeOperation() {
+        this.likeType = null;
```

Comment:
same question as above. SupportsShowLikeOperation is a common abstract class for many showClasses. it must offer a default constructor, otherwise may change the hive-parser.

Reply:
hive parser module has been dropped in master. Besides, your construct are same as the default init behavior. I'm not quite get `otherwise may change the hive-parser`.

Reply:
Sorry for misleading, No problem with this comments now.